Service/DocumentService.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+
+namespace EMS\CoreBundle\Service;
+
+
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use EMS\CoreBundle\Elasticsearch\Bulker;
+use EMS\CoreBundle\Entity\ContentType;
+use EMS\CoreBundle\Entity\Context\DocumentImportContext;
+use EMS\CoreBundle\Entity\Revision;
+use EMS\CoreBundle\Exception\CantBeFinalizedException;
+use EMS\CoreBundle\Form\Form\RevisionType;
+use Symfony\Component\Form\FormFactoryInterface;
+
+class DocumentService
+{
+
+    /** @var DataService */
+    protected $dataService;
+    /** @var FormFactoryInterface */
+    private $formFactory;
+    /** @var Registry */
+    private $doctrine;
+    /** @var Bulker */
+    private $bulker;
+    /** @var string */
+    private $instanceId;
+
+    public function __construct(Registry $doctrine, DataService $dataService, FormFactoryInterface $formFactory, Bulker $bulker, string $instanceId)
+    {
+        $this->dataService = $dataService;
+        $this->formFactory = $formFactory;
+        $this->doctrine = $doctrine;
+        $this->bulker = $bulker;
+        $this->instanceId = $instanceId;
+    }
+
+    public function initDocumentImporterContext(ContentType $contentType, string $lockUser, bool $rawImport, bool $signData, bool $indexInDefaultEnv, int $bulkSize, bool $finalize, bool $force) : DocumentImportContext
+    {
+        $entityManager = $this->doctrine->getManager();
+        $this->bulker->setEnableSha1(false);
+        $this->bulker->setSize($bulkSize);
+        return new DocumentImportContext($entityManager, $contentType, $lockUser, $rawImport, $signData, $indexInDefaultEnv, $finalize, $force);
```

Comment:
the context should contain a state. It should not contain other services --> entityManager should be part of the DocumentService only. The context should only have Getter methods. No functions (unless it transforms the inputs without depending on other services)

Reply:
bulksize and enable sha1 should also be part of the context